EZP-31190 Document using the URLService - URLs filtered by Section (#940)
ezsystems/developer-documentation: docs/api/public_php_api_managing_repository.md

$languageCreateStruct->name = 'German';
$this->languageService->createLanguage($languageCreateStruct);
```

## URLs

### URLService

[`URLService`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/URLService.php)

> enables you to find, load and update URLs.

Reviewer: ```suggestion
enables you to find, load and update external URLs used in ezrichtext and ezurl field types.
```

> enables you to find, load and update external URLs used in ezrichtext and ezurl field types.

Reviewer 2: ```suggestion
enables you to find, load and update external URLs used in RichText and URL Fields.
```

To view a list of all URLs, use [`URLService::findUrls`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/URLService.php#L33)

`URLService::findUrls` takes as argument a [`URLQuery`,](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/Values/URL/URLQuery.php)
in which you need to specify:

> - query filter e.g. section

Reviewer 2: ```suggestion
- query filter e.g. Section
```

> - sort clauses for URL queries

Reviewer 2: ```suggestion
- Sort Clauses for URL queries
```

- offset for search hits, used for paging the results

> - query limit, if value is `0`, search query will not return any search hits

Reviewer 2: ```suggestion
- query limit. If value is `0`, search query will not return any search hits
```

```php
$URLQuery = $this->URLService->newURLQuery();
$URLQuery->filter = 'section';
$URLQuery->sortClauses = 'descending';
$URLQuery->offset = '0';
$URLQuery->limit = '25'

> $this->URLService->findUrls($query);

Reviewer: ```php
use eZ\Publish\API\Repository\Values\URL\Query\Criterion;
use eZ\Publish\API\Repository\Values\URL\Query\SortClause; 

# ...

$urlQuery = $this->urlService->newURLQuery();
$urlQuery->filter = new Criterion\LogicalAnd([
    new Criterion\SectionIdentifier('media'),
    new Criterion\Validity(true),
]);
$urlQuery->sortClauses = [
   new SortClause\URL(SortClause::SORT_DESC)
];
$urlQuery->offset = 0;
$urlQuery->limit = 25;

$this->urlService->findUrls($query);
```
Please test this code snippet before commit!
